cmd/copy.go

```diff
@@ -17,79 +16,141 @@ import (
 var getSessionIDRegex = regexp.MustCompile(`^sess:([^/]+)`)
 
 func Copy(cmd *cobra.Command, args []string) error {
-	scpArgs := make([]string, 0, len(args))
-	var targetExec *types.Exec
+	exec, err := getTargetExec(cmd, args)
+	if err != nil {
+		return err
+	}
+
+	if exec == nil {
+		return fmt.Errorf("At least one remote host must be specified")
+	}
+	if exec.Connection == nil {
+		return fmt.Errorf("Target session must have an active connection")
+	}
+	if exec.SSHKey.PublicKey == nil && config.SSHPublicKeyPath == "" {
+		return fmt.Errorf("Failed to identify public key, check your Unweave config file or specify it manually")
+	}
+
+	scpArgs, err := formatPaths(exec, args)
+	if err != nil {
+		ui.Infof("❌ Unsuccessful copy: %s", err.Error())
+	}
+	ui.Infof(fmt.Sprintf("🔄 Copying %s => %s", scpArgs[0], scpArgs[1]))
+
+	switch {
+	case isLocalDirToRemoteCopy(args):
```

Comment:
The args should be pars in this function and not passed to handlers as a slice. Here, we know the exact number of args expected because cobra validated them for us. If you pass them down to smaller private functions as a slice, we lose the information about the slice length. Another function could later call the `isLocalDirToRemoteCopy` function with an empty slice an cause a panice.

Instead, something like this is much safer and clearer to understand:

```go
isLocalDirToRemoteCopy(pathA, pathB string)
```